internal/reporting/markdown.go

```diff
@@ -0,0 +1,342 @@
+			continue
+		}
+
+		if test.Attempt.Status.Kind == v1.TestStatusTimedOut {
+			testsBySection[timedOutSection] = append(testsBySection[timedOutSection], test)
+			continue
+		}
+
+		if test.Attempt.Status.Kind == v1.TestStatusQuarantined {
+			testsBySection[quarantinedSection] = append(testsBySection[quarantinedSection], test)
+			continue
+		}
+
+		if test.Attempt.Status.Kind == v1.TestStatusCanceled {
+			testsBySection[canceledSection] = append(testsBySection[canceledSection], test)
+			continue
+		}
+	}
+
+	return testsBySection
+}
+
+func writeMarkdownFlakySection(markdown *strings.Builder, tests []v1.Test) (bool, error) {
+	return writeMarkdownSection(
+		markdown,
+		flakySection,
+		tests,
+		func(test v1.Test) *v1.TestStatus {
+			if test.Attempt.Status.PotentiallyFlaky() {
+				return &test.Attempt.Status
+			}
+
+			for _, attempt := range test.PastAttempts {
+				if attempt.Status.PotentiallyFlaky() {
+					return &attempt.Status
+				}
+			}
+
+			return nil
+		},
+	)
+}
+
+func writeMarkdownFailedSection(markdown *strings.Builder, tests []v1.Test) (bool, error) {
+	return writeMarkdownSection(
+		markdown,
+		failedSection,
+		tests,
+		func(test v1.Test) *v1.TestStatus {
+			return &test.Attempt.Status
+		},
+	)
+}
+
+func writeMarkdownTimedOutSection(markdown *strings.Builder, tests []v1.Test) (bool, error) {
+	return writeMarkdownSection(
+		markdown,
+		timedOutSection,
+		tests,
+		func(test v1.Test) *v1.TestStatus {
+			return &test.Attempt.Status
+		},
+	)
+}
+
+func writeMarkdownQuarantinedSection(markdown *strings.Builder, tests []v1.Test) (bool, error) {
+	return writeMarkdownSection(
+		markdown,
+		timedOutSection,
+		tests,
+		func(test v1.Test) *v1.TestStatus {
+			return test.Attempt.Status.OriginalStatus
+		},
+	)
+}
+
+func writeMarkdownCanceledSection(markdown *strings.Builder, tests []v1.Test) (bool, error) {
+	return writeMarkdownSection(
+		markdown,
+		timedOutSection,
+		tests,
+		func(test v1.Test) *v1.TestStatus {
+			return &test.Attempt.Status
+		},
+	)
+}
+
+func writeMarkdownSection(
+	markdown *strings.Builder,
+	section markdownTestSection,
+	tests []v1.Test,
+	findFailedStatus func(v1.Test) *v1.TestStatus,
+) (bool, error) {
+	if len(tests) == 0 {
+		return false, nil
+	}
+
+	if _, err := markdown.WriteString(fmt.Sprintf("\n## %v\n\n", section)); err != nil {
+		return false, errors.WithStack(err)
+	}
+
+	parsedTemplate, err := template.New("markdownTestTemplate").Parse(markdownTestTemplate)
+	if err != nil {
+		return false, errors.WithStack(err)
+	}
+
+	for _, test := range tests {
+		location := ""
+		if test.Location != nil {
+			location = test.Location.String()
+		}
+
+		failedStatus := findFailedStatus(test)
+		markdownTest := markdownTest{
+			Name:     test.Name,
+			Location: location,
+			Retries:  len(test.PastAttempts),
+		}
+		if failedStatus != nil {
+			markdownTest.Message = failedStatus.Message
+			markdownTest.Backtrace = strings.Join(failedStatus.Backtrace, "\n")
+		}
```

Comment:
Once we have the substitutions and some configuration in here, we can build the retry command and pass it through